test/src/test/java/org/corfudb/infrastructure/log/StreamLogFilesTest.java

```diff
@@ -422,4 +422,19 @@ public void testPrefixTrim() {
         // Address 0 is not reflected in trimAddress
         assertThat(trimmedExceptions).isEqualTo(trimAddress + 1);
     }
+
+    @Test
+    public void testPrefixTrimAndStartUp() {
+        StreamLog log = new StreamLogFiles(getContext(), false);
+        log.prefixTrim(StreamLogFiles.RECORDS_PER_LOG_FILE / 2);
+        log.compact();
```

Comment:
Without the code fix, the first 3 lines would cause the 4th line to throw an exception which will cause the test to fail. It just simulates a logunit restart after a compaction run.

So, if you remove the code fix (not the first 3 lines), the test wouldnt pass.